ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala

```diff
@@ -49,7 +49,20 @@ object domain {
 
   type LfValue = lf.value.Value[lf.value.Value.ContractId]
 
-  case class JwtPayload(ledgerId: LedgerId, applicationId: ApplicationId, party: Party)
+  // Until we get multi-party submissions, write endpoints require a single party.
+  case class JwtWritePayload(ledgerId: LedgerId, applicationId: ApplicationId, party: Party) {
+    def toReadPayload: JwtPayload = JwtPayload(ledgerId, applicationId, List(), List(party))
+  }
+
+  // JWT payload that preserves readAs and actAs and supports multiple parties. This is currently only used for
+  // read endpoints but once we get multi-party submissions, this can also be used for write endpoints.
+  case class JwtPayload(
+      ledgerId: LedgerId,
+      applicationId: ApplicationId,
+      readAs: List[Party],
+      actAs: List[Party]) {
```

Comment:
If it is not allowed for either of these to be empty, I would prefer to use non-empty types; this applies throughout the modeling of party sets in this PR, for which e.g. you might use `OneAnd[Set, ?]` or some newtype that enforces the invariant.